File: doc/src/scripts/IR.cpp
#include <krpc.hpp>
#include <krpc/services/infernal_robotics.hpp>
#include <iostream>
#include <vector>

using namespace krpc::services;

int main() {
  auto conn = krpc::connect("InfernalRobotics Example");
  InfernalRobotics infernal_robotics(&conn);

  InfernalRobotics::ServoGroup group = infernal_robotics.servo_group_with_name("MyGroup");
  if (group == InfernalRobotics::ServoGroup())
    std::cout << "Group not found" << std::endl;

  std::vector<InfernalRobotics::Servo> servos = group.servos();
  for (auto servo : servos)
    std::cout << servo.name() << " " << servo.position() << std::endl;

  group.move_right();
  sleep(1);
  group.stop();
}

File: doc/src/scripts/IR.cs
using KRPC.Client;
using KRPC.Client.Services.SpaceCenter;
using KRPC.Client.Services.InfernalRobotics;
using System;
using System.Threading;
using System.Net;

class IR {
    public static void Main () {
        var connection = new Connection (name: "InfernalRobotics Example");
        var vessel = connection.SpaceCenter ().ActiveVessel;
        var ir = connection.InfernalRobotics ();

        var group = ir.ServoGroupWithName (vessel, "MyGroup");
        if (group == null) {
            Console.WriteLine ("Group not found");
            return;
        }

        foreach (var servo in group.Servos)
            Console.WriteLine (servo.Name + " " + servo.Position);

        group.MoveRight ();
        Thread.Sleep (1000);
        group.Stop ();
    }
}
